### PR TITLE
Tftp vxlan backport/v4

### PR DESCRIPTION
**Pull request by shivan1b**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
- https://redmine.openinfosecfoundation.org/issues/3044
- https://redmine.openinfosecfoundation.org/issues/3008
- https://redmine.openinfosecfoundation.org/issues/3068

### CODE REVIEW

File: src/decode-vxlan.c [hunk 1]
```diff
@@ -268,9 +268,11 @@ static int DecodeVXLANtest02 (void)
 void DecodeVXLANRegisterTests(void)
 {
 #ifdef UNITTESTS
-    UtRegisterTest("DecodeVXLANtest01",
-                   DecodeVXLANtest01);
-    UtRegisterTest("DecodeVXLANtest02",
-                   DecodeVXLANtest02);
+	if (g_vxlan_enabled == true) {
```
Comment (victorjulien) [inline src/decode-vxlan.c]: This is not how this is supposed to be handled. The tests should be registered unconditionally, and then the tests themselves can temporary update the global ```g_vxlan_enabled``` to true to run as normal. Be sure to reset them at the end.

Comment (victorjulien) [reply src/decode-vxlan.c]: indent was off too it seems

Git commit message should be updated to something like "decoder/vxlan: disable by default"

Comment (shivan1b) [reply src/decode-vxlan.c]: Oh OK.

**Comment by victorjulien** [comment]
I merged all commits except the vxlan ones through #4013. Please submit a new PR with the vxlan patches. Thanks!

**Pull request closed without merging**